### PR TITLE
feat(auth): Add OIDC support types & docs

### PR DESCRIPTION
**Pull request by palpatim**
## Issue \#
N/A

## Description
* Add docs for OIDC support types
* Add TaskQueue test &docs
* Add default no-op reset method for Plugins

## General Checklist

- [X] Added new tests to cover change, if needed
- [X] Build succeeds with all target using Swift Package Manager
- [X] ~~All~~ New unit tests pass
- [ ] ~~All integration tests pass~~
- [ ] ~~Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)~~
- [X] Documentation update for the change if required
- [X] PR title conforms to conventional commit style
- [x] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] ~~If breaking change, documentation/changelog update with migration instructions~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
+//
```
Comment (palpatim) [inline AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift:1]: I'm not actually sure this type belongs in AWSPluginsCore -- open to moving it

**Comment by palpatim** [comment]
TaskQueue test is failing b/c of Xcode version mismatch. Will talk to @aws-amplify/amplify-ios team to see right way to address.

File: AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift (hunk 1 above)
Comment (thisisabhash) [reply AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift:1]: This seems like the correct place - other Auth definitions are also defined in this folder.

File: AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/// the authentication provider, the state of the authentication tokens, and even whether the network is reachable. Types
+/// that conform to ``AWSAuthSessionBehavior`` must implement the `isSignedIn` flag to conform to the
+/// following rules:
+/// TODO: Add detailed discussion of isSignedIn for Cognito. Include discussion of cases left deliberately up to the discretion
+/// of the plugin developer.
+public protocol AWSAuthSessionBehavior<OIDCCredentials> : AuthSession {
+
+    /// The concrete type holding the OIDC tokens from the authentication provider. Generally, this type will have at least
+    /// methods for retrieving an identity token and an access token.
+    associatedtype OIDCCredentials
```
Comment (thisisabhash) [inline AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift:29]: how about we rename this to `TokensType`

File: AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    /// TODO: What errors can result here?
+    var awsCredentialsResult: Result<AWSTemporaryCredentials, AuthError> { get }
+
+    /// The result of the most recent attempt to get a
+    /// [Cognito identity pool identity ID](https://docs.aws.amazon.com/cognitoidentity/latest/APIReference/API_GetId.html#CognitoIdentity-GetId-response-IdentityId).
+    /// The identityID may represent either an unauthenticated or authenticated identity, depending on the configuration of the
+    /// identity pool and the tokens used to retrieve the identity ID from Cognito.
+    ///
+    /// If the most recent attempt caused an error, the result will contain the details of the error.
+    /// TODO: What errors can result here?
```
Comment (thisisabhash) [inline AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift:46]: AuthError: https://github.com/aws-amplify/amplify-swift/blob/main/Amplify/Categories/Auth/Error/AuthError.swift

File: AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift (hunk 2 above)
Comment (palpatim) [reply AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift:29]: Counteroffer: `Tokens` ?

Comment (thisisabhash) [reply AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthSessionBehavior.swift]: Sounds good

**Review by thisisabhash — APPROVED** [review]

**Review by thisisabhash — APPROVED** [review]

**Pull request merged**